app/views/assessments/_partial_feedback_js.html.erb

```diff
@@ -0,0 +1,66 @@
+<%= external_javascript_include_tag "lodash" %>
+<script>
+  function refreshPartial(clicked) {
+    $.ajax({
+        url: `getPartialFeedback?job_id=${<%= @job_id %>}`,
+        success: (res) => {
+          if (!res) return;
+          if (res.partial_feedback) {
+            $('#partial-feedback').html(`
+              <pre>${res.partial_feedback}</pre>
+              <div class="autograding-in-progress">
+                <i class="refresh-feedback small material-icons" onclick="rotateIcon(); onClickRefresh()">refresh</i>
+                <pre>Autograding in progress</pre>
+              </div>`);
+            $('.feedback-status').html("In-Progress");
+            $('.feedback-status').removeClass("feedback-status__queued");
+            $('.feedback-status').addClass("feedback-status__inprogress");
+            $( "#result-body" ).load(window.location.href + " #result-body" );
+
+          } else if (res.queue_position != null) {
+            $('#partial-feedback').html(`
+              <div class="autograding-in-progress">
+                <pre>You are at position ${(res.queue_position) + 1} / ${(res.queue_length)} in the autograding queue.</pre>
```

Comment:
Nit: Do students need to know / care about the `queue_length`? Perhaps displaying only `queue_position` would be sufficient.